### PR TITLE
chore(test) Make a shorter alias for django_db with params

### PR DESCRIPTION
**Pull request by markstory**
Follow up from #51552 to address feedback on needing a shorter alias for managing django's database state now that doing so takes more effort.

### CODE REVIEW

**Review by corps — APPROVED** [review]

**Review by wedamija — APPROVED** [review]

File: src/sentry/utils/pytest/fixtures.py [hunk 1]
```diff
@@ -139,6 +139,18 @@
 }
 
 
+def django_db_all(func=None, *, transaction=None, **kwargs):
+    """Pytest decorator for resetting all databases"""
+
+    if func is not None:
+        return pytest.mark.django_db(func)
```
Comment (jjbayer) [inline src/sentry/utils/pytest/fixtures.py:146]: I would have expected this to also get `databases="__all__"`, but I might be missing something.

**Review by jjbayer — APPROVED** [review]
Nice!

**Review by iambriccardo — APPROVED** [review]

File: src/sentry/utils/pytest/fixtures.py (hunk 1 above)
Comment (markstory) [reply src/sentry/utils/pytest/fixtures.py:146]: You're totally right.

File: src/sentry/utils/pytest/fixtures.py [hunk 2]
```diff
@@ -139,6 +139,18 @@
 }
 
 
+def django_db_all(func=None, *, transaction=None, **kwargs):
```
Comment (asottile-sentry) [inline src/sentry/utils/pytest/fixtures.py:142]: this can just be `django_db_all = pytest.mark.django_db(...)` I believe

Comment (markstory) [reply src/sentry/utils/pytest/fixtures.py:142]: I gave that a shot with 

```diff
diff --git a/src/sentry/utils/pytest/fixtures.py b/src/sentry/utils/pytest/fixtures.py
index ba8f4cfc38..2f79a8315c 100644
--- a/src/sentry/utils/pytest/fixtures.py
+++ b/src/sentry/utils/pytest/fixtures.py
@@ -139,17 +139,7 @@ DEFAULT_EVENT_DATA = {
 }
 
 
-def django_db_all(func=None, *, transaction=None, **kwargs):
-    """Pytest decorator for resetting all databases"""
-
-    if func is not None:
-        return pytest.mark.django_db(transaction=transaction, databases="__all__")(func)
-
-    def decorator(function):
-        return pytest.mark.django_db(transaction=transaction, databases="__all__")(function)
-
-    return decorator
-
+django_db_all = pytest.mark.django_db(..., databases="__all__")
 
 @pytest.mark.django_db
 @pytest.fixture
```

but decorator usage like `@django_db_all(transaction=True)` failed with 


```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/_pytest/main.py", line 270, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/_pytest/main.py", line 323, in _main
INTERNALERROR>     config.hook.pytest_collection(session=session)
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/_pytest/main.py", line 334, in pytest_collection
INTERNALERROR>     session.perform_collect()
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/_pytest/main.py", line 667, in perform_collect
INTERNALERROR>     hook.pytest_collection_modifyitems(
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pytest_django/plugin.py", line 402, in pytest_collection_modifyitems
INTERNALERROR>     items.sort(key=get_order_number)
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pytest_django/plugin.py", line 387, in get_order_number
INTERNALERROR>     transaction = validate_django_db(marker_db)[0]
INTERNALERROR>   File "/Users/markstory/code/sentry/.venv/lib/python3.8/site-packages/pytest_django/plugin.py", line 756, in validate_django_db
INTERNALERROR>     return apifun(*marker.args, **marker.kwargs)
INTERNALERROR> TypeError: apifun() got multiple values for argument 'transaction'
```

I think there are a few competing layers of metaprogramming colliding here. I can revisit streamlining this definition in the future though.

**Pull request merged**